### PR TITLE
Issue 1625 - Option for Windows-compatible filenames - trim trailing spaces and dots

### PR DESCRIPTION
**Pull request by dgolda**
Fix for: https://tvheadend.org/issues/1625#note-5

### CODE REVIEW

File: src/dvr/dvr_rec.c [hunk 1]
```diff
@@ -151,7 +151,8 @@ cleanup_filename(char *s, dvr_config_t *cfg)
   if (s[0] == '.')
     s[0] = '_';
 
-  for (i = 0, len = strlen(s); i < len; i++) {
+  int len2 = strlen(s);
```
Comment (perexg) [inline src/dvr/dvr_rec.c]: Move this declaration to the declaration block (start of function).

File: src/dvr/dvr_rec.c [hunk 2]
```diff
@@ -169,6 +170,18 @@ cleanup_filename(char *s, dvr_config_t *cfg)
       s[i] = '_';
   }
 
+  if(cfg->dvr_windows_compatible_filenames) {
+    //trim trailing spaces and dots
+    for (i = len2 - 1; i >= 0; i--) {
+      if((s[i] == ' ') || (s[i] == '.')) {
```
Comment (perexg) [inline src/dvr/dvr_rec.c]: Could you rewrite this condition as opposite:

if (!wrong_char)
  break;
s[i] = '\0';

It's more clean.

File: src/dvr/dvr_rec.c (hunk 1 above)
Comment (dgolda) [inline src/dvr/dvr_rec.c]: Hmm, strange convention, but OK. Sure.

File: src/dvr/dvr_rec.c (hunk 2 above)
Comment (dgolda) [inline src/dvr/dvr_rec.c]: OK

**Comment by perexg** [comment]
Merged. Thanks.

**Pull request closed without merging**